gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -369,288 +339,362 @@ public static void setMetaData(final FileHandler fileHandler, final Photo photo,
+  private MediaUtil() {
   }
 
   /**
-   * Return the written file
-   * @param image the image from which the resolution must be red
-   * @param outputFile the output file
-   * @param widthParam the width parameter
-   * @param watermark the height parameter
-   * @param nameWatermark the watermark to write on the output file
-   * @param sizeWatermark the size of the watermark
-   * @throws Exception
+   * In charge of processing an internal media.
+   * @param <M>
    */
-  private static void resizePhoto(final BufferedImage image, final HandledFile outputFile,
-      final int widthParam, final boolean watermark, final String nameWatermark,
-      final int sizeWatermark) throws Exception {
-    OutputStream outputStream = null;
-    try {
-      outputStream = outputFile.openOutputStream();
-      final ImageResizer resizer = new ImageResizer(image, widthParam);
-      if (watermark) {
-        resizer.resizeImageWithWatermark(outputStream, nameWatermark, sizeWatermark);
-      } else {
-        resizer.resizeImage(outputStream);
-      }
-    } finally {
-      if (outputStream != null) {
-        IOUtils.closeQuietly(outputStream);
-      }
-    }
-  }
-
-  private static void createWatermark(final OutputStream watermarkedTargetStream,
-      final String watermarkLabel, final BufferedImage image, final int percentSizeWatermark)
-      throws IOException {
+  private static abstract class MediaProcess<M extends InternalMedia> {
 
-    final int imageWidth = image.getWidth();
-    final int imageHeight = image.getHeight();
+    private final HandledFile handledFile;
+    private final M media;
+    private final Set<MediaMimeType> supportedMimeTypes;
 
-    // création du buffer a la même taille
-    final BufferedImage outputBuf =
-        new BufferedImage(imageWidth, imageHeight, BufferedImage.TYPE_INT_RGB);
+    private MediaMimeType physicalFileMimeType = null;
+    private org.silverpeas.core.io.media.MetaData physicalFileMetaData = null;
 
-    final double max = Math.max(imageWidth, imageHeight);
-
-    // recherche de la taille du watermark en fonction de la taille de la photo
-    int size = 8;
-    if (max < 600) {
-      size = 8;
+    private MediaProcess(final HandledFile handledFile, final M media) {
+      this.handledFile = handledFile;
+      this.media = media;
+      this.supportedMimeTypes = MediaMimeType.getSupportedMimeTypes(media.getType());
     }
-    if (max >= 600 && max < 750) {
-      size = 10;
+
+    /**
+     * Processes the media files.
+     * @throws Exception
+     */
+    public void process() throws Exception {
+      try {
+        setInternalMetadata();
+        generateFiles();
+      } finally {
+        close();
+      }
     }
-    if (max >= 750 && max < 1000) {
-      size = 12;
+
+    /**
+     * Generates specific media files.
+     * @throws Exception
+     */
+    protected abstract void generateFiles() throws Exception;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)